### PR TITLE
[15.0][MIG] Migration of module sale_operating_unit

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

Migration PR superseded PR https://github.com/OCA/operating-unit/pull/479
It also contains the fix of PR https://github.com/OCA/operating-unit/pull/510

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

**Comment by alan196** [comment]
@BT-apadron  @AaronHForgeFlow @MiquelRForgeFlow Could you help me to review this PR 🙏

**Review by BT-aleonard — APPROVED** [review]
Thanks for migrating, 
Looks good to me!

**Review by BT-anieto — COMMENTED** [review]
Hello @alan196 ! Thx for migrating this module

The code LGTM, but there are some cosmetic changes that you should apply:

1) Follow the guide in https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests#mergesquash-the-commits-generated-by-bots-or-weblate to squash administrative commits
2) Your [MIG] commit should only have migration changes. I'm a bit surprised that you're changing the index.html.

File: sale_operating_unit/views/sale_view.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+<odoo>
+    <record id="view_quotation_tree" model="ir.ui.view">
+        <field name="name">sale.order.tree</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_quotation_tree" />
+        <field name="arch" type="xml">
+            <field name="user_id" position="after">
+                <field
+                    name="operating_unit_id"
+                    domain="[('user_ids', 'in', uid)]"
```
Comment (Saran440) [inline sale_operating_unit/views/sale_view.xml]:
```suggestion
                    
```

File: sale_operating_unit/views/sale_view.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    </record>
+    <record id="view_order_tree" model="ir.ui.view">
+        <field name="name">sale.order.tree</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_tree" />
+        <field name="arch" type="xml">
+            <field name="user_id" position="after">
+                <field
+                    name="operating_unit_id"
+                    domain="[('user_ids', 'in', uid)]"
```
Comment (Saran440) [inline sale_operating_unit/views/sale_view.xml]:
```suggestion
                   
```

File: sale_operating_unit/views/sale_view.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <field name="user_id" position="after">
+                <field
+                    name="operating_unit_id"
+                    widget="selection"
+                    domain="[('user_ids', 'in', uid)]"
```
Comment (Saran440) [inline sale_operating_unit/views/sale_view.xml]:
```suggestion
                    
```

**Review by Saran440 — COMMENTED** [review]
Code review.
Please remove all domain user in operating unit.

**Comment by AaronHForgeFlow** [comment]
@alan196 , can you rebase to rebuild runboat? also agree to remove the domains

**Comment by alan196** [comment]
@AaronHForgeFlow @Saran440 Rebased and with the domains removed

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review.

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review.

**Comment by AaronHForgeFlow** [comment]
/ocabot merge nobump

Merging as comments attended

**Pull request merged**